src/Illuminate/Queue/RedisQueue.php

```diff
@@ -133,11 +133,16 @@ public function pop($queue = null)
             $this->migrateAllExpiredJobs($queue);
         }
 
-        $job = $this->getConnection()->lpop($queue);
+        $script = <<<'LUA'
```

Comment:
It lpops a job from the main queue, if exists. Then it adds the job to the
reserved sorted set with the appropriate score (current time + expire). The
logic is exactly as before, plus the safety of transaction atomicity.
However, I think I will change the logic to be more like what is done in
database driver. By that, I mean to add attempts counter before reserving
the job. It requires json manupulation inside lua script, which fortunately
is guaranteed to be supported by cjson library.
On Apr 1, 2016 7:09 PM, "Taylor Otwell" <notifications@github.com> wrote:

> In src/Illuminate/Queue/RedisQueue.php
> <https://github.com/laravel/framework/pull/12655#discussion_r58214736>:
>
> > @@ -133,11 +133,16 @@ public function pop($queue = null)
> >              $this->migrateAllExpiredJobs($queue);
> >          }
> >
> > -        $job = $this->getConnection()->lpop($queue);
> > +        $script = <<<'LUA'
>
> Can you describe what this Lua script does?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/laravel/framework/pull/12655/files/e76fa889c5e4b012b30a111537c6fcaab37c0504#r58214736>
>